lib/buffer.js

```diff
@@ -380,8 +379,7 @@ function fromArrayBuffer(obj, byteOffset, length) {
   } else {
     // convert length to non-negative integer
     length = +length;
-    // Check for NaN
-    if (length !== length) {
+    if (isNaN(length)) {
```

Comment:
Wouldn't we want `Number.isNaN()` in these cases?

Reply:
Seems reasonable, since we already converted to number with `+`. Thanks!

Reply:
Well actually, that seems to slow some things down, with more confidence than plain `isNaN`.

```
                                                     improvement confidence     p.value
 buffers\\adjust-offset.js n=1000000 input="float"      -2.91 %         ** 0.009562388
 buffers\\adjust-offset.js n=1000000 input="inf"         1.80 %            0.582680085
 buffers\\adjust-offset.js n=1000000 input="large"      -1.33 %            0.700712702
 buffers\\adjust-offset.js n=1000000 input="medium"     -3.29 %          * 0.017033448
 buffers\\adjust-offset.js n=1000000 input="nan"        -1.63 %            0.603059991
 buffers\\adjust-offset.js n=1000000 input="one"        -4.68 %         ** 0.003543366
 buffers\\adjust-offset.js n=1000000 input="zero"        0.49 %            0.754381672
```

Since `Number.isNaN` is the clearest method here and seems to have a negative impact, I'm closing this PR.